File: Builder/src/com/company/Tour.java
package com.company;

import java.util.ArrayList;

public class Tour {
    private int days;
    private int count;
    private String hotel;
    private ArrayList<Program> tourProgram;

    public ArrayList<Program> getTourProgram() {
        return tourProgram;
    }

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }

    public int getDays() {
        return days;
    }

    public String getHotel() {
        return hotel;
    }

    public void setDays(int days) {
        this.days = days;
    }

    public void setHotel(String hotel) {
        this.hotel = hotel;
    }
    public void addProgram(Program pr) {
        tourProgram.add(pr);
    }

    public Tour() {
        hotel = null;
        days = 1;
        count = 1;
    }
}

File: Builder/tour.cpp
#include "tour.h"
#include <iostream>
Tour::Tour()
{

}

void Tour::setDays(int d)
{
 days = d;
}
int Tour::getDays()
{
    return days;
}
void Tour::addProgrammDay(Programm *p)
{
 programm.push_back(*p);
}
void Tour::info()
{
    double sum=0;
    std::cout << "tour for " << count << " persons consist is "<< days << " days" << std::endl;
    std::cout << "tour programm: "<< std::endl;
    for (auto pd : programm)
    {
        auto activities = pd.getActivities();
        for (auto act : activities)
        {
            std::cout << act << std::endl;
        }
        sum += pd.getTotal();
    }
    std::cout << "programm cost:  "<< sum*count<< std::endl;
}

std::string Tour::getHotel()
{
    return hotel;
}
void Tour::setHotel(std::string h)
{
    hotel = h;
}
double Tour::getHotelCost()
{
    return this->hotelPerDay*count*days;
}
void Tour::setHotelPerDay(double cost)
{
    hotelPerDay = cost;
}
void Tour::setCount(int cnt)
{
    count = cnt;
}
